java/format/pom.xml

```diff
@@ -23,6 +23,10 @@
 <name>Arrow Format</name>
 <description>Generated Java files from the IPC Flatbuffer definitions.</description>
 
+  <properties>
+    <project.build.outputTimestamp>2023-12-13T00:00:00Z</project.build.outputTimestamp>
+  </properties>
```

Comment:
Can you provide some commands or steps for validating a reproducible build, such as in the Arrow Format module?

Reply:
When I tried these commands, it finished ok with or without any changes:

````
$ mvn clean install
$ mvn clean verify artifact:compare
...
[INFO] Saved info on build to /Users/dsusanibar/fork/arrow/java/format/target/arrow-format-15.0.0-SNAPSHOT.buildinfo
[INFO] Checking against reference build from central...
[INFO] Reference buildinfo file not found: it will be generated from downloaded reference artifacts
[INFO] Reference build java.version: 11 (from MANIFEST.MF Build-Jdk-Spec)
[INFO] Reference build os.name: Unix (from pom.properties newline)
[INFO] Minimal buildinfo generated from downloaded artifacts: /Users/dsusanibar/fork/arrow/java/format/target/reference/arrow-format-15.0.0-SNAPSHOT.buildinfo
[INFO] Reproducible Build output summary: 5 files ok
[INFO] Reproducible Build output comparison saved to /Users/dsusanibar/fork/arrow/java/format/target/arrow-format-15.0.0-SNAPSHOT.buildcompare
...
````

Reply:
You are right for the check: mvn clean install and then mvn clean verify artifact:compare do a check. 
This PR ensures we have reproducible build, not meaning that some module are not ok already (AFAIR, flight and algorithm were not reproducible).

Reply:
You can also check with artifact:check-buildplan.

Reply:
What we might want is to add this to CI somewhere (or possibly the release verification process) so we can make sure it stays working.

Reply:
@lidavidm good idea. Let me check to add a workflow in GH actions.